lib/archivist/index.js

```diff
@@ -774,7 +774,13 @@ Archivist.prototype.set = function (topic, index, data, mediaType, encoding, exp
 	trackSize(topic, index, value);
 
 	value.set(mediaType, data, encoding);
-	value.touch(expirationTime);
+
+	if (!expirationTime) {
+		const config = configuration.getTopicConfig(topic);
+		value.touch(Date.now() / 1000 + config.ttl);
```

Comment:
What will happen if config.ttl is not set?

Reply:
Right, I will modify it

Reply:
For add, do you mean touch ?

Reply:
No, there should be an `archivist.add` normally.

Reply:
I don't see any archivist.add

Reply:
Ok, might be some higher level stuff. Don't worry about it.